GateKeeper.cpp

```diff
@@ -383,6 +385,105 @@ namespace pEp {
         return enabled;
     }
 
+    bool GateKeeper::is_update_available() {
+        bool result = true;
+        NTSTATUS status = BCryptOpenAlgorithmProvider(&hAES, BCRYPT_AES_ALGORITHM, MS_PRIMITIVE_PROVIDER, 0);
+        assert(status == 0);
+        result = status == 0;
+        if (status)
+            goto closing;
+        status = BCryptSetProperty(hAES, BCRYPT_CHAINING_MODE, (PUCHAR)BCRYPT_CHAIN_MODE_GCM, sizeof(BCRYPT_CHAIN_MODE_GCM), 0);

```

Comment:
there is a bug here using your approach. Imagine that at this point result is true as a result of the previous comparison and now status is returned as status == 1. This means something went wrong and it failed so the goto closing is executed and you will be returning result == true saying that an update is available which is not correct, instead result should be false in this case as something went wrong or failed.

Reply:
done